### PR TITLE
Releases [ TS-59 ]

### PR DESCRIPTION
**Pull request by zepod**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -16,6 +16,7 @@
     // eslint-plugin-import
     "import/no-unresolved": [2, {"commonjs": true}],
     "import/named": 2,
+    "curly": 0,
```
Comment (zepod) [inline .eslintrc:19]:
```
if (bool)
   return value;
```
is sexy syntax

Comment (kepi74) [reply .eslintrc:19]: I personally prefer to write all conditions the same way. This just adds some noice in code for readers.

File: app/react/configs/lib/store/createDependencyInjections.js [hunk 2]
```diff
@@ -7,11 +7,13 @@ export default function createDependencyInjections() {
...
       dispatchAndThrow,
       dispatchPromise,
+      genericInterface: createInterface(''),
       projectsInterface: createInterface('projects'),
       localesInterface: createInterface('locales'),
       hierarchyInterface: createInterface('hierarchy'),
       keysInterface: createInterface('keys'),
-      translationsInterface: createInterface('translations')
+      translationsInterface: createInterface('translations'),
+      releasesInterface: createInterface('releases')
```
Comment (kepi74) [inline app/react/configs/lib/store/createDependencyInjections.js:16]: What about to condense interfaces to one dependency key?

```javascript
const interfaces = {
  genericInterface: createInterface(''),
  ...
};
...
withDispatch: {
  dispatchAndThrow,
  dispatchPromise,
  interfaces,
}

File: app/react/projects/list/components/Locale.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 /* @flow */
 import React from 'react';
 import { Flex, Box } from 'radium-flex';
+import { Link } from 'react-router';
 import locale from 'country-language';
+
 import { calculatePercents } from '../../helpers';
 
 import type { LocaleEntityType } from '../../../types/entityTypes';
```
Comment (kepi74) [inline app/react/projects/list/components/Locale.js]: It's not good idea to have Component props type outside Component itself, I think.

File: app/react/projects/list/components/Locale.js [hunk 4]
```diff
@@ -10,16 +12,24 @@ export default class Locale extends React.PureComponent {
   props: LocaleEntityType
 
   render() {
-    const { code, translationCount, translatedCount } = this.props;
+    const { projectId, code, translationCount, translatedCount, id } = this.props;
```
Comment (kepi74) [inline app/react/projects/list/components/Locale.js]: I'm suggesting to rename `id` to `localeId`, just to be consistent with naming of different IDs in this component.

File: app/react/releases/actions.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+export const CREATE_RELEASE_FULFILLED = 'CREATE_RELEASE_FULFILLED';
+
+export const TOGGLE_APPROVE_KEY = 'TOGGLE_APPROVE_KEY';
+
+const flatMethod = res => res.json().then(data => ({
+  translations: CamelCase(data.translations, { deep: true }),
+  hierarchy: data.hierarchy
+}));
+
+const getIds = (state: Object, { localeId, projectId }: Object): Array<string> => {
```
Comment (kepi74) [inline app/react/releases/actions.js]: We don't have state type declaration?

File: app/react/releases/reducer.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+export default function reducer(state = initialState, action: Action) {
+  switch (action.type) {
+    case FETCH_NOT_APPROVED_HIERARCHY_PENDING:
+      return {
+        ...state,
+        pending: true
+      };
+    case FETCH_NOT_APPROVED_HIERARCHY_FULFILLED: {
```
Comment (kepi74) [inline app/react/releases/reducer.js:21]: Missing case for rejected request.

File: app/react/releases/transform.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+const maxLevel = 10;
+const transformHierarchy = (structure: Object, level: number = 0, path = []): Array<Object> => {
+  if (level > maxLevel) {
+    // eslint-disable-next-line no-console
+    console.warn('Provided hierarchy object was to deep. Either provide different object or increase "maxLevel" variable');
```
Comment (kepi74) [inline app/react/releases/transform.js]: 👍

**Review by kepi74 — CHANGES_REQUESTED** [review]
Add handling of fetch errors to release reducer, pls.

**Pull request merged**